### PR TITLE
Allow other applications to use GA4 code

### PR DESCRIPTION
**Pull request by andysellick**
## What / why
Extends the existing `load-analytics.js` to allow other domain details to be included when checking how to initialise GA4 (and UA, although this is soon to be deprecated). This is part of the work needed to allow the components gem analytics code to run on our publishing applications (we can also switch to using this method for the dev docs analytics, instead of hard coding the details here).

## Visual Changes
None.

### CODE REVIEW

**Review by AshGDS — APPROVED** [review]
LGTM 👍 Few small things, and does it need a CHANGELOG?

File: app/assets/javascripts/govuk_publishing_components/load-analytics.js [hunk 1]
```diff
@@ -70,6 +70,19 @@ window.GOVUK.loadAnalytics = {
...
   ],
 
+  loadExtraDomains: function () {
+    if (this.extraDomainsAreValid(window.GOVUK.analyticsGa4Domains)) {
+      this.domains = this.domains.concat(window.GOVUK.analyticsGa4Domains)
+    }
+  },
+
+  extraDomainsAreValid: function (domains) {
+    if (!domains || !Array.isArray(domains)) {
```
[the remark this answers is not in the document]
Comment (AshGDS) [reply app/assets/javascripts/govuk_publishing_components/load-analytics.js:80]: May not need a function for this if that's the case

File: spec/javascripts/govuk_publishing_components/load-analytics.spec.js [hunk 2]
```diff
@@ -138,6 +138,44 @@ describe('Analytics loading', function () {
       expect(window.GOVUK.analyticsGa4.vars).toEqual(null)
       window.GOVUK.loadAnalytics.domains[0].initialiseGA4 = true
     })
+
+    describe('when addition domain details are needed', function () {
```
[the remark this answers is not in the document]
Comment (andysellick) [reply spec/javascripts/govuk_publishing_components/load-analytics.spec.js]: Whoops, fixed, thanks.

File: app/assets/javascripts/govuk_publishing_components/load-analytics.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/govuk_publishing_components/load-analytics.js]: Good point, thanks! And yeah, definitely doesn't need a function. I'd written it this way as I was originally going to make `extraDomainsAreValid` check in more detail that the array contained valid information, but then decided against it. Have removed.

**Review by AshGDS — APPROVED** [review]
LGTM 👍

**Pull request merged**